src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public void add(T value) {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T data, Node<T> prev, Node<T> next) {
+            this.data = data;
+            this.prev = prev;
+            this.next = next;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        addLast(value);
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (index == size) {
+            addLast(value);
+        } else {
+            Node<T> nodeAtIndex = getNode(index);
+            Node<T> newNode = new Node<>(value, nodeAtIndex.prev, nodeAtIndex);
+            if (nodeAtIndex.prev != null) {
+                nodeAtIndex.prev.next = newNode;
+            } else {
+                first = newNode;
+            }
+            nodeAtIndex.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).data;
     }
 
-    @Override
-    public T set(T value, int index) {
-        return null;
+    private Node<T> getNode(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToRemove = getNode(index);
+        if (nodeToRemove.prev != null) {
+            nodeToRemove.prev.next = nodeToRemove.next;
+        }
+        if (nodeToRemove.next != null) {
+            nodeToRemove.next.prev = nodeToRemove.prev;
+        }
+        if (nodeToRemove == first) {
+            first = nodeToRemove.next;
+        }
+        if (nodeToRemove == last) {
+            last = nodeToRemove.prev;
+        }
```

Comment:
![image](https://github.com/mate-academy/jv-linked-list/assets/48316721/8cb7f4a7-fbc4-447c-a88e-30de51618fb4)
Be more attentive to requirements